src/main/java/spoon/reflect/visitor/ImportScannerImpl.java

```diff
@@ -448,7 +448,9 @@ protected boolean addMethodImport(CtExecutableReference ref) {
 	protected boolean isImportedInMethodImports(CtExecutableReference<?> ref) {
 		if (!(ref.isImplicit()) && methodImports.containsKey(ref.getSimpleName())) {
 			CtExecutableReference<?> exist = methodImports.get(ref.getSimpleName());
-			if (exist.getSignature().equals(ref.getSignature())) {
+			if ((exist.getDeclaringType() != null ? exist.getDeclaringType().getQualifiedName() : "" + "." + exist.getSignature()).equals(
```

Comment:
could you add brackets around `(x?y:z)`, so the precedence of operators is clear even for me ;-)
Or better to create private (or public?) method `getQualifiedSignature()'